server/channels/app/channel_bookmark.go

```diff
@@ -130,3 +134,51 @@ func (a *App) UpdateChannelBookmarkSortOrder(bookmarkId, channelId string, newIn
 
 	return bookmarks, nil
 }
+
+func (a *App) GetChannelsWithBookmarksForSession(c request.CTX, session *model.Session, channels model.ChannelList, since int64) []*model.ChannelWithBookmarks {
```

Comment:
If an error happens in any of these functions, the user would get a response (albeit without bookmarks, but with valid channel info) and it would not be managed nor logged. I think we should return and manage the error normally, as with this logic it would almost 100% sure be related to a database issue that we wouldn't want to get unnoticed